crates/ruff_macros/src/combine_options.rs

```diff
@@ -48,13 +48,53 @@ fn handle_field(field: &Field) -> syn::Result<proc_macro2::TokenStream> {
             ..
         }) => match segments.first() {
             Some(PathSegment {
-                ident: type_ident, ..
-            }) if type_ident == "Option" => Ok(quote_spanned!(
-                ident.span() => #ident: self.#ident.or(other.#ident)
-            )),
+                ident: type_ident,
+                arguments,
+            }) if type_ident == "Option" => {
```

Comment:
I prefer to have "dumb" derive macros that, ideally, simply call a trait method rather than containing the entire implementation. 

This is achievable for `CombinePluginOptions`, although it requires a bit more boilerplate code. 

`handle_field` in macro:

```rust
    Ok(quote_spanned!(
        ident.span() => #ident: crate::configuration::CombinePluginOptions::combine(self.#ident, other.#ident)
    ))
```

next to `CombinePluginOptions`

```
macro_rules! or_combine_plugin_impl {
    ($ty:ident) => {
        impl CombinePluginOptions for Option<$ty> {
            #[inline]
            fn combine(self, other: Self) -> Self {
                self.or(other)
            }
        }
    };
}

or_combine_plugin_impl!(bool);
or_combine_plugin_impl!(u8);
or_combine_plugin_impl!(u16);
or_combine_plugin_impl!(u32);
or_combine_plugin_impl!(u64);
or_combine_plugin_impl!(usize);
or_combine_plugin_impl!(i8);
or_combine_plugin_impl!(i16);
or_combine_plugin_impl!(i32);
or_combine_plugin_impl!(i64);
or_combine_plugin_impl!(isize);
or_combine_plugin_impl!(String);

impl<T: CombinePluginOptions> CombinePluginOptions for Option<T> {
    fn combine(self, other: Self) -> Self {
        match (self, other) {
            (Some(base), Some(other)) => Some(base.combine(other)),
            (Some(base), None) => Some(base),
            (None, Some(other)) => Some(other),
            (None, None) => None,
        }
    }
}

impl<T> CombinePluginOptions for Option<Vec<T>> {
    fn combine(self, other: Self) -> Self {
        self.or(other)
    }
}

impl<T, S> CombinePluginOptions for Option<HashSet<T, S>> {
    fn combine(self, other: Self) -> Self {
        self.or(other)
    }
}

impl<K, V, S> CombinePluginOptions for HashMap<K, V, S>
where
    K: Eq + Hash,
    S: BuildHasher,
{
    fn combine(mut self, other: Self) -> Self {
        self.extend(other);
        self
    }
}

or_combine_plugin_impl!(ParametrizeNameType);
or_combine_plugin_impl!(ParametrizeValuesType);
or_combine_plugin_impl!(ParametrizeValuesRowType);
or_combine_plugin_impl!(Quote);
or_combine_plugin_impl!(Strictness);
or_combine_plugin_impl!(RelativeImportsOrder);
or_combine_plugin_impl!(LineLength);
or_combine_plugin_impl!(Convention);
or_combine_plugin_impl!(IndentStyle);
or_combine_plugin_impl!(QuoteStyle);
or_combine_plugin_impl!(LineEnding);
or_combine_plugin_impl!(DocstringCodeLineWidth);
```

We could avoid all the boilerplate if Rust supported specialization...  (it would boil down to `impl<T> CombinePluginOptions for Option<T>` and one specialization `impl<T: CombinePluginOptions> for Option<T>`). Maybe @BurntSushi knows a better way to avoid some of the boilerplate. 

The advantage of being explicit about how `CombinePluginOptions` is implemented is that we can have custom implementations (like for HashMap) or even other types that we don't know of yet. It also ensures that we explicitly think about how `Option<T>` should be merged rather than assuming `Option::or` is the right choice.

Reply:
Sounds good, I can do this.

Reply:
I applied the changes for you